community/kernel/src/main/java/org/neo4j/kernel/impl/api/LookupFilter.java

```diff
@@ -110,7 +110,8 @@ public boolean test( long nodeId )
         @Override
         Property nodeProperty( long nodeId, int propertyKeyId ) throws EntityNotFoundException
         {
-            return readOperations.nodeGetProperty( state, nodeId, propertyKeyId );
+            Object value = readOperations.nodeGetProperty( state, nodeId, propertyKeyId );
+            return value == null ? Property.noNodeProperty( nodeId, propertyKeyId ) : Property.property( propertyKeyId, value );
```

Comment:
Problem is that we need valueEquals from Property. To remove these Property objects all of that would have to be worked into a separate class